library/bignum_core.h

```diff
@@ -293,6 +293,21 @@ int mbedtls_mpi_core_write_be(const mbedtls_mpi_uint *A,
 void mbedtls_mpi_core_shift_r(mbedtls_mpi_uint *X, size_t limbs,
                               size_t count);
 
+/**
+ * \brief               Shift an MPI in-place left by a number of bits.
+ *
+ *                      Shifting by more bits than there are bit positions
```

Comment:
I don't see how this is undefined behaviour. The code as it is will still zero the MPI, the same as shift r

Reply:
When I said undefined behaviour I didn't mean undefined behaviour as per the C standard. I meant that we should call out in the documentation of the function, that whatever we do in those cases, might not make sense and can change at any time. I suggested to add that in the documentation, because if it is there, we don't need to update the documentation every time we change the behaviour.

Reply:
Ah - perhaps something along the lines of "The result of shifting by more bits than there are bit positions is undefined" then? - using the phrase "undefined behaviour" is dangerous here I think (it clearly triggered me). We would need to update the comment on shift-r as well, potentially (not in this PR)

Reply:
I expect “undefined behavior” to mean “never do this: anything might happen”, and “unspecified result” to mean “do this only if you don't care about the result”. The language in the C standard distinguishes between “undefined” (anything can happen) and “unspecified” (the behavior is constrained but not fully), and it's easier for readers if we use the same distinction in our documentation.

I would not particularly expect left and right shift to have the same behavior: it's well understood that a right shift destroys the rightmost bits, whereas one might expect a left shift on a bignum (as opposed to a fixnum) to preserve the leftmost bits. Extrapolating from that to the case where all bits should be destroyed, that case could well be different for a left shift and a right shift.